src/scheduler/node_partition.cpp

```diff
@@ -916,16 +869,13 @@ find_alloc_np_cache(status *policy, np_cache ***pnpc_arr,
 			if (sort_func != NULL)
 				qsort(nodepart, num_parts, sizeof(node_partition *), sort_func);
 
-			npc = new_np_cache();
+			npc = new np_cache();
```

Comment:
npc will never be null, as new will raise an exception if it fails.

do you want to put a try/catch block here?